cf-agent/verify_packages.c

```diff
@@ -201,7 +201,7 @@ PromiseResult VerifyPackagesPromise(EvalContext *ctx, const Promise *pp)
             break;
         case PACKAGE_PROMISE_TYPE_OLD:
             Log(LOG_LEVEL_VERBOSE,
-                "Using old package promise. Please note that this will be obsolete in 3.8");
+                "Using old package promise. Please note that the old implementation is considered obsolete as of 3.8. The old implementation will continue to work, but forward development will be directed toward the new implementation.");
```

Comment:
The nice thing about the old package method is that we do need external modules for the package installation.  With the external modules you create dependencies on the language it is written in. For now it's python and there can also be problems with version python version 2 and 3.  The executable python can be installed in different locations.  

I agree with @kacfengine with the new package module there is only support for yum/apt.  My colleague that asked the question about windows was worried about the statement that his package statements were deprecated.